test/cppsuite/test_harness/workload/thread_context.cxx

```diff
@@ -285,6 +285,45 @@ thread_context::insert(
     return (true);
 }
 
+bool
```

Comment:
`insert` and `remove` functions are very very similar, should we find a way to combine them?

Reply:
I see both sides to this. I believe `insert` and `remove` are very fundamentally different operations. Possibly combining both functions might be quite confusing when reading the code. I am fine with both ways here.

Reply:
I think separate is better here.